internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,378 @@
+					// We ignore it because its coming out of our own process.
+					Tls:           tls,
+					TlsSkipVerify: true,
+				},
+			}),
+		)
+		if err != nil {
+			log.Error("trigger connection back to gRPC failed", "err", err)
+			return
+		}
+		defer grpcConn.Close()
+
+		// Our API client
+		client := pb.NewWaypointClient(grpcConn)
+
+		requestVars := mux.Vars(r)
+		runTriggerId := requestVars["id"]
+
+		overrideVarJSONRaw := r.URL.Query().Get("variables")
+		var (
+			vo                map[string]string
+			variableOverrides []*pb.Variable
+		)
+
+		if overrideVarJSONRaw != "" {
+			if err := json.Unmarshal([]byte(overrideVarJSONRaw), &vo); err != nil {
+				http.Error(w, fmt.Sprintf("failed to decode 'variables' json request param into a map: %s", err), 500)
+				return
+			}
+
+			for name, value := range vo {
+				v := &pb.Variable{
+					Name:   name,
+					Source: &pb.Variable_Cli{Cli: &empty.Empty{}},
+				}
+
+				if valBool, err := strconv.ParseBool(value); err == nil {
+					v.Value = &pb.Variable_Bool{Bool: valBool}
+				} else if valInt, err := strconv.ParseInt(value, 10, 64); err == nil {
+					v.Value = &pb.Variable_Num{Num: valInt}
+				} else {
+					// NOTE: for this case, it can either be a "string" or
+					// complex HCL type like an array or map. We can set this value
+					// as a Variable_Str here, and when we go to parse the variables
+					// later we do the proper string versus HCL check in variables.go
+					v.Value = &pb.Variable_Str{Str: value}
+				}
+
+				variableOverrides = append(variableOverrides, v)
+			}
+		}
+
+		var resp *pb.RunTriggerResponse
+		runTriggerReq := &pb.RunTriggerRequest{
+			Ref: &pb.Ref_Trigger{
+				Id: runTriggerId,
+			},
+			VariableOverrides: variableOverrides,
+		}
+
+		// attempt to make a grpc request to run trigger by id
+		if requireAuth {
+			resp, err = client.RunTrigger(ctx, runTriggerReq)
+		} else {
+			resp, err = client.NoAuthRunTrigger(ctx, runTriggerReq)
+		}
+		if err != nil {
+			log.Error("server failed to run trigger", "id", runTriggerId, "err", err)
+
+			if status.Code(err) == codes.PermissionDenied {
+				http.Error(w, fmt.Sprintf("request not authorized to run trigger: %s", err), 401)
+			} else {
+				// improve http error code, which is more applicable for general queue failures?
+				http.Error(w, fmt.Sprintf("server failed to run trigger: %s", err), 412)
+			}
+
+			return
+		}
+		if resp == nil {
+			http.Error(w, fmt.Sprintf("server returned no job ids from run trigger %q", runTriggerId), 500)
+			return
+		}
+		jobIds := resp.JobIds
+
+		streamOutput := r.URL.Query().Get("stream")
+
+		// Attempt to stream output back, on request.
+		if streamOutput != "" {
+			if !requireAuth {
+				// We do not allow streaming job stream info if a no-auth token trigger was requested
+				log.Trace("server does not allow for streaming job stream output for no-token trigger URLs")
+				w.WriteHeader(http.StatusNoContent)
```

Comment:
This specific status code was a request/feedback from @mitchellh. I think it matches how Netlify handles no-auth triggers. An unauthorized status could make sense here too 🤔

Reply:
I think based on your other comment feedback, we can return `http.StatusNoContent` if no streaming was requested, and otherwise if a stream was requested for a non-authenticated token, we return `http.StatusUnauthorized`. 🤔

Reply:
I suppose I didn't or don't quite understand the flow here. If the scenario is that you can trigger this without a token but require a token to receive the stream, then `StatusNoContent` makes sense if there is no token present. I think the name and purpose of `requireAuth` has thrown me a bit here.